pexec/managed_process.go

```diff
@@ -265,6 +268,12 @@ func (p *managedProcess) manage(stdOut, stdErr io.ReadCloser) {
 	default:
 	}
 
+	// Run onCrashHandler if it exists. Do not attempt restart if onCrashHandler
+	// returns false.
+	if p.onCrashHandler != nil && !p.onCrashHandler() {
```

Comment:
I think `onUnexpectedExit` would be nicer. Also using `on` obviates the need for `Handler`.

Reply:
Can (should?) we pass the exit code in?

Reply:
Changed to `OnUnexpectedExit` and made the signature `func(int) bool` to have access to exit code. Would be nice for exit handler to better understand what happened to the crashed process.

Reply:
definitelyl